packages/catalog-model/src/EntityPolicies.ts

```diff
@@ -51,12 +54,11 @@ class AllEntityPolicies implements EntityPolicy {
 class AnyEntityPolicy implements EntityPolicy {
   constructor(private readonly policies: EntityPolicy[]) {}
 
-  async enforce(entity: Entity): Promise<Entity> {
+  async enforce(entity: Entity): Promise<Entity | undefined> {
```

Comment:
This interface change is here because the old behaviour was bonkers - we tried all validators and thew away the exceptions, hoping that at least one would match. That means you'd get a completely nondescript message when that happened because we didn't know which one of the exception messages to show. Now the validators only apply to the exact type of entity that they know how to handle, and ignore the rest, so we can make exceptions fatal instead of sweeping them under the rug.

Reply:
That is less bonkers :p